### PR TITLE
Prepare gensim v3.8.2 to pin smart_open for Py2.7 compatibility

### LINKED ISSUE #2786: Recent versions of smart-open (1.11.0 and 1.11.1)  break gensim in Python 2.7
### Problem description

What are you trying to achieve? What is the expected result? What are you seeing instead?

Using gensim with Python2.7

#### Steps/code/corpus to reproduce
```
pip install gensim
...
python
>>> import gensim
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/brukau/tmp/gensim/local/lib/python2.7/site-packages/gensim/__init__.py", line 5, in <module>
    from gensim import parsing, corpora, matutils, interfaces, models, similarities, summarization, utils  # noqa:F401
  File "/home/brukau/tmp/gensim/local/lib/python2.7/site-packages/gensim/parsing/__init__.py", line 4, in <module>
    from .preprocessing import (remove_stopwords, strip_punctuation, strip_punctuation2,  # noqa:F401
  File "/home/brukau/tmp/gensim/local/lib/python2.7/site-packages/gensim/parsing/preprocessing.py", line 42, in <module>
    from gensim import utils
  File "/home/brukau/tmp/gensim/local/lib/python2.7/site-packages/gensim/utils.py", line 45, in <module>
    from smart_open import open
  File "/home/brukau/tmp/gensim/local/lib/python2.7/site-packages/smart_open/__init__.py", line 28, in <module>
    from .smart_open_lib import open, parse_uri, smart_open, register_compressor
  File "/home/brukau/tmp/gensim/local/lib/python2.7/site-packages/smart_open/smart_open_lib.py", line 23, in <module>
    import pathlib
ImportError: No module named pathlib
>>> 
```

#### Versions

Please provide the output of:

```python
import platform; print(platform.platform())
import sys; print("Python", sys.version)
import numpy; print("NumPy", numpy.__version__)
import scipy; print("SciPy", scipy.__version__)
import gensim; print("gensim", gensim.__version__)
from gensim.models import word2vec;print("FAST_VERSION", word2vec.FAST_VERSION)
```

Linux-5.3.0-45-generic-x86_64-with-Ubuntu-19.10-eoan
('Python', '2.7.17 (default, Nov  7 2019, 10:07:09) \n[GCC 9.2.1 20191008]')
('NumPy', '1.16.1')
('SciPy', '1.2.3')
gensim import fails

### PR DESCRIPTION
**Pull request by brukau**
Pin version of smart-open to <1.11.

Pull request to master as soon as gensim develop seems to be Python3 only.

Fixes #2786.

### CODE REVIEW

**Comment by brukau** [comment]
cc @mpenkov

File: setup.py [hunk 1]
```diff
@@ -395,7 +395,7 @@ def finalize_options(self):
         NUMPY_STR,
         'scipy >= 0.18.1',
         'six >= 1.5.0',
-        'smart_open >= 1.8.1',
+        'smart_open >= 1.8.1, < 1.11',
```
Comment (piskvorky) [inline setup.py:398]:
```suggestion
        'smart_open >= 1.8.1, < 1.11',  # smart_open >= 1.11 is py3+ only. TODO: Remove once we drop py2.7 from gensim too.
```

File: setup.py [hunk 2]
```diff
@@ -395,7 +395,7 @@ def finalize_options(self):
         NUMPY_STR,
         'scipy >= 0.18.1',
         'six >= 1.5.0',
-        'smart_open >= 1.8.1',
+        'smart_open >= 1.8.1, < 1.11',  # smart_open >= 1.11 is py3+ only. TODO: Remove once we drop py2.7 from gensim too.
```
Comment (mpenkov) [inline setup.py:398]: We've already dropped Py2.7 support for gensim on the develop branch. So, this change is not required.

**Comment by menshikh-iv** [comment]
@mpenkov sounds good for me, but please fix appveyor before that

**Comment by mpenkov** [comment]
Thank you for taking care of Appveyor @menshikh-iv !!!

**Pull request merged**